lib/web_ui/test/canvaskit/canvas_golden_test.dart

```diff
@@ -0,0 +1,390 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.12
+import 'dart:html' as html;
+import 'dart:math' as math;
+import 'dart:typed_data';
+
+import 'package:test/bootstrap/browser.dart';
+import 'package:test/test.dart';
+import 'package:ui/src/engine.dart';
+import 'package:ui/ui.dart' as ui;
+
+import 'package:web_engine_tester/golden_tester.dart';
+
+import 'common.dart';
+
+void main() {
+  internalBootstrapBrowserTest(() => testMain);
+}
+
+const ui.Rect region = const ui.Rect.fromLTRB(0, 0, 500, 250);
+
+Future<void> matchPictureGolden(String goldenFile, CkPicture picture, { bool write = false }) async {
+  final EnginePlatformDispatcher dispatcher = ui.window.platformDispatcher as EnginePlatformDispatcher;
+  final LayerSceneBuilder sb = LayerSceneBuilder();
+  sb.pushOffset(0, 0);
+  sb.addPicture(ui.Offset.zero, picture);
+  dispatcher.rasterizer!.draw(sb.build().layerTree);
+  await matchGoldenFile(goldenFile, region: region, write: write);
+}
+
+void testMain() {
+  group('CkCanvas', () {
+    setUpCanvasKitTest();
+
+    test('can restore a picture from a snapshot', () async {
```

Comment:
Please add a test that shows that no recording canvas is created when `browserSupportsFinalizationRegistry = true`

Reply:
Done.